### PR TITLE
  systemd: New bootc-fetch-apply-updates.{timer,service} 

### PR DESCRIPTION
**Pull request by cgwalters**
Let's ship a baseline systemd unit that can be enabled for automatic updates.

### CODE REVIEW

File: systemd/bootc-apply.service [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Apply bootc updates
+Documentation=man:bootc(8)
+ConditionPathExists=/run/ostree-booted
+
+[Service]
+Type=oneshot
+ExecStart=bootc update --apply --quiet
```
Comment (vrothberg) [inline systemd/bootc-apply.service:8]:
```suggestion
ExecStart=/usr/bin/bootc update --apply --quiet
```

File: systemd/bootc-apply.service [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
```
Comment (vrothberg) [inline systemd/bootc-apply.service:9]:
```suggestion
```

File: systemd/bootc-apply.service (hunk 1 above)
Comment (vrothberg) [reply systemd/bootc-apply.service:8]: Unless there's a reason not to, I prefer absolute paths to be explicit.

File: systemd/bootc-apply.service
Comment (jlebon) [inline systemd/bootc-apply.service:1]: Minor: maybe `bootc-apply-updates.{service,timer}`? Feels like `apply` isn't descriptive enough.

Comment (jlebon) [reply systemd/bootc-apply.service:1]: WFM!

**Comment by cgwalters** [comment]
OK per https://github.com/containers/bootc/pull/179#discussion_r1383761082 let's think about this one a bit more.  We should perhaps also have `bootc-fetch.service` as an explicitly separated unit?

While this project is still classified as "interfaces subject to change", these particular units will be a really important "UX touchpoint" and it'd be good to get them as close to right as we think early on.

This whole thread really intersects with https://github.com/containers/bootc/issues/2 which is a tension I've felt since the start.

**Comment by vrothberg** [comment]
> OK per [#179 (comment)](https://github.com/containers/bootc/pull/179#discussion_r1383761082) let's think about this one a bit more. We should perhaps also have `bootc-fetch.service` as an explicitly separated unit?

I wonder when such a unit would be useful?  Splitting fetching updates from applying them seems special.

If there's a use case to check for an update prior to executing it, callers could do an `upgrade --check` and then fire the unit.

PR as is LGTM

**Comment by jlebon** [comment]
> > OK per [#179 (comment)](https://github.com/containers/bootc/pull/179#discussion_r1383761082) let's think about this one a bit more. We should perhaps also have `bootc-fetch.service` as an explicitly separated unit?
> 
> I wonder when such a unit would be useful? Splitting fetching updates from applying them seems special.

With [finalization locking](https://github.com/ostreedev/ostree/pull/1841), you can do all the steps ahead of time (fetch and deploy) and minimize down time to just updating the bootloader and restarting.

**Comment by cgwalters** [comment]
Right, for locking let's reference https://github.com/ostreedev/ostree/issues/3025

**Comment by cgwalters** [comment]
Actually one thing I'm really strongly opinionated on is that we should default to respecting systemd *blocking* inhibitors; see https://github.com/coreos/rpm-ostree/pull/2862

It allows the use case of `ssh root@examplemachine systemd-inhibit --mode=block bash` to debug something reliably.

I am also thinking here that alongside it'd make sense to add an explicit `bootc-apply.service` (and CLI verb that starts it?) that can have arbitrary hooks added.

For example, zincati today has code to invoke the classic [notify via tty](https://github.com/coreos/zincati/pull/479) code.  It's not clear to me that we should duplicate that in bootc, but we could easily support it externally via code that adds a systemd-drop-in that adds `ExecStartPre` or so?

**Comment by cgwalters** [comment]
OK, did https://github.com/ostreedev/ostree/pull/3090

**Comment by rhatdan** [comment]
Is this ready to merge?

**Comment by cgwalters** [comment]
> Actually one thing I'm really strongly opinionated on is that we should default to respecting systemd blocking inhibitors; see https://github.com/coreos/rpm-ostree/pull/2862

We still don't do this...definitely in the nice to have.

**Comment by cgwalters** [comment]
We're also not using finalization locking here yet.  Not fatal, because we're doing an "all-in-one" unit that does everything in one go, so the case of handling out-of-cycle reboots is a bit moot.

**Pull request merged**